File: 26.删除排序数组中的重复项.cpp
#include<iostream>
#include<string>
#include<vector>
using namespace std;


class Solution {
public:
	int removeDuplicates(vector<int>& nums) {
		if (nums.size() == 0)
		{
			return 0;
		}
		auto i = 0;
		auto j = 1;

		while (j < nums.size())
		{
			if (nums[i] == nums[j])
			{
				j++;
			}
			else
			{
				nums[++i] = nums[j++];
			}
		}
		return i + 1;
		
	}
};

int main()
{
	string temp;
	vector<int> nums = vector<int>(0);
	cout << nums.size();
	Solution s = Solution();
	cout << s.removeDuplicates(nums);

	cin >> temp;
	return 0;
}

File: 26.删除排序数组中的重复项.py
class Solution:
    def removeDuplicates(self, nums):
        nums = list(set(nums))
        return len(nums)

# class Solution:
#     def removeDuplicates(self, nums):
#         i = 0
#         while(i<(len(nums)-1)):
#             if(nums[i] == nums[i+1]):
#                 del nums[i+1]
#                 continue
#             else:
#                 i += 1
#         return len(nums)

if __name__ == "__main__":
    s = Solution()
    print(list(set([1,1,2])))

